### PR TITLE
Feature/upgrade node to release ash

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 1]
```diff
@@ -1166,25 +1163,88 @@ components:
...
-            - ecdsa_secp256k1
-            - eddsa_ed25519
-            - xrd
+            - genesis_helper
+            - faucet
           properties:
-            faucet:
-              $ref: "#/components/schemas/ComponentAddress"
+            xrd:
+              type: string
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1195]: Shouldn't we use `$ref: ".../Address"`?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs [hunk 2]
```diff
@@ -281,281 +281,297 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
...
+                            CoreModel.EntityType.GlobalVirtualEd25519Identity => typeof(GlobalIdentityEntity),
+                            CoreModel.EntityType.GlobalVirtualSecp256k1Identity => typeof(GlobalIdentityEntity),
+                            CoreModel.EntityType.GlobalFungibleResource => typeof(GlobalFungibleResourceEntity),
+                            CoreModel.EntityType.GlobalNonFungibleResource => typeof(GlobalNonFungibleResourceEntity),
+                            _ => throw new ArgumentOutOfRangeException(nameof(newGlobalEntity.EntityType), newGlobalEntity.EntityType.ToString()),
+                        });
+                    }
 
-                    if (substateData is CoreModel.EpochManagerSubstate epochManagerSubstate)
+                    foreach (var substate in stateUpdates.CreatedSubstates.Concat(stateUpdates.UpdatedSubstates).ToList())
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:342]: What's the point of `.ToList()`?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs [hunk 3]
```diff
@@ -1084,22 +1135,21 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
                     return ret;
                 });
 
-                var nonFungibleIdStore = nonFungibleIdStoreHistoryToAdd.GetOrAdd(new NonFungibleStoreLookup(e.ReferencedStore.DatabaseGlobalAncestorId, e.StateVersion), _ =>
+                var nonFungibleIdStore = nonFungibleIdStoreHistoryToAdd.GetOrAdd(new NonFungibleStoreLookup(e.ReferencedResource.DatabaseId, e.StateVersion), _ =>
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:1138]: There's no more concept of NF Store, right?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs [hunk 4]
```diff
@@ -1353,26 +1401,27 @@ void AggregateEntityResourceVaultInternal(long entityId, long resourceEntityId,
...
                         VaultEntityId = e.ReferencedVault.DatabaseId,
-                        IsRoyaltyVault = e.ReferencedVault.GetDatabaseEntity<VaultEntity>().IsRoyaltyVault,
+                        IsRoyaltyVault = e.ReferencedVault.GetDatabaseEntity<InternalFungibleVaultEntity>().IsRoyaltyVault,
                         Balance = e.Balance,
                     };
                 })
                 .ToList();
 
             var entityNonFungibleVaultHistoryToAdd = nonFungibleVaultChanges
+                .GroupBy(x => new { x.StateVersion, x.ReferencedVault, x.ReferencedResource })
```
Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:1411]: Why do we need this GroupBy?

Comment (KrzysztofLabus-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:1411]: I'm just thinking if we shouldn't use this "previous + mostRecentXxx" pattern. Are we sure we can operate only on data seen in TX stream? i.e. don't we need previous record form database?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs [hunk 5]
```diff
@@ -218,27 +217,21 @@ private static void HookupEntities(ModelBuilder modelBuilder)
...
+            .HasValue<InternalAccountEntity>(EntityType.InternalAccountComponent)
+            .HasValue<GlobalAccountEntity>(EntityType.GlobalAccountComponent)
+            .HasValue<GlobalPackageEntity>(EntityType.GlobalPackage)
+            .HasValue<InternalKeyValueStoreEntity>(EntityType.InternalKeyValueStore)
+            .HasValue<InternalFungibleVaultEntity>(EntityType.InternalFungibleVault)
+            .HasValue<InternalNonFungibleVaultEntity>(EntityType.InternalNonFungibleVault)
+            .HasValue<GlobalClockEntity>(EntityType.GlobalClock)
+            .HasValue<GlobalValidatorEntity>(EntityType.GlobalValidator)
+            .HasValue<GlobalAccessControllerEntity>(EntityType.GlobalAccessController)
+            .HasValue<GlobalIdentityEntity>(EntityType.GlobalIdentity);
```
[the remark this answers is not in the document]
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/CommonDbContext.cs:234]: fixed.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs (hunk 2 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs:342]: removed. I think i just forgot to remove it after debugging.

**Pull request merged**